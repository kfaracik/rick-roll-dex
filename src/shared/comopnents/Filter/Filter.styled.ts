import {StyleSheet} from 'react-native';
import {Colors} from '../../utils';

export const styles = StyleSheet.create({
  container: {
    marginBottom: 16,
  },
  card: {
    marginBottom: 16,
    marginTop: 8,
  },
  filterButton: {
    marginHorizontal: 16,
  },
  buttonsContainer: {
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-between',
  },
  categoryName: {
    fontFamily: 'DMMono-Regular',
    color: Colors.mediumGreen,
    fontSize: 14,
  },
});
