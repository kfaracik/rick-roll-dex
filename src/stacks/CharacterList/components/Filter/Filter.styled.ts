import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  card: {
    marginBottom: 32,
    marginTop: 16,
  },
  buttonsContainer: {
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-between',
  },
});
