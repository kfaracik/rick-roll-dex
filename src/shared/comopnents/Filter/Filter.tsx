import React, {useState} from 'react';
import {ScrollView, View, Text} from 'react-native';
import {styles} from './Filter.styled';
import {Species, Status} from '../../../shared/api';
import {Button} from '../Button';
import {Colors} from '../../utils';
import {Card} from '../Card';
import {CheckboxItem} from '../CheckboxItem';
import Icon from 'react-native-vector-icons/AntDesign';
import {AnimatedChevron} from '../AnimatedChevron';

type FilterProps = {
  onApplyFilters: (status: Status, species: Species) => void;
};

export const Filter = ({onApplyFilters}: FilterProps) => {
  const [filterOptionsExpanded, setFilterOptionsExpanded] = useState(false);
  const [status, setStatus] = useState<Status>(null);
  const [species, setSpecies] = useState<Species>(null);

  const handleStatusChange = (value: Status) => {
    setStatus(value === status ? null : value);
  };

  const handleSpeciesChange = (value: Species) => {
    setSpecies(value === species ? null : value);
  };

  const onApplyPress = () => {
    setFilterOptionsExpanded(false);
    onApplyFilters(status, species);
  };

  const onResetFiltersPress = () => {
    setStatus(null);
    setSpecies(null);
  };

  const onFilterButtonPress = () => {
    setFilterOptionsExpanded(prev => !prev);
  };

  return (
    <ScrollView style={styles.container}>
      <Button
        title="FILTER "
        onPress={onFilterButtonPress}
        mode="primary"
        style={styles.filterButton}>
        <AnimatedChevron
          isOpen={filterOptionsExpanded}
          accessibilityLabel={'show filters'}
        />
      </Button>
      {filterOptionsExpanded && (
        <Card style={styles.card}>
          <View>
            <Text style={styles.categoryName}>STATUS</Text>
            <CheckboxItem
              label="Alive"
              value="Alive"
              status={status}
              onPress={handleStatusChange}
            />
            <CheckboxItem
              label="Dead"
              value="Dead"
              status={status}
              onPress={handleStatusChange}
            />
            <CheckboxItem
              label="Unknown"
              value="Unknown"
              status={status}
              onPress={handleStatusChange}
            />
          </View>
          <View>
            <Text style={styles.categoryName}>SPECIES</Text>
            <CheckboxItem
              label="Human"
              value="Human"
              status={species}
              onPress={handleSpeciesChange}
            />
            <CheckboxItem
              label="Humanoid"
              value="Humanoid"
              status={species}
              onPress={handleSpeciesChange}
            />
          </View>
          <View style={styles.buttonsContainer}>
            <Button title="RESET" onPress={onResetFiltersPress} mode="white" />
            <Button title="APPLY" onPress={onApplyPress} mode="primary" />
          </View>
        </Card>
      )}
    </ScrollView>
  );
};
