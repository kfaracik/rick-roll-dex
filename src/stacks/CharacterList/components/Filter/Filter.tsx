import React, {useState} from 'react';
import {ScrollView, View, Text} from 'react-native';
import {Button, Card} from '../../../../shared/comopnents';
import {Colors} from '../../../../shared/utils';
import {styles} from './Filter.styled';
import {CheckboxItem} from '../CheckboxItem';
import {Species, Status} from '../../../../shared/api';

type FilterProps = {
  onApplyFilters: (status: Status, species: Species) => void;
};

export const Filter = ({onApplyFilters}: FilterProps) => {
  const [filterOptionsExpanded, setFilterOptionsExpanded] = useState(false);
  const [status, setStatus] = useState<Status>(null);
  const [species, setSpecies] = useState<Species>(null);

  const handleStatusChange = (value: string) => {
    setStatus(value === status ? null : value);
  };

  const handleSpeciesChange = (value: string) => {
    setSpecies(value === species ? null : value);
  };

  const onApplyPress = () => {
    setFilterOptionsExpanded(false);
    onApplyFilters(status, species);
  };

  const onResetFiltersPress = () => {
    setStatus(null);
    setSpecies(null);
  };

  const onFilterButtonPress = () => {
    setFilterOptionsExpanded(prev => !prev);
  };

  return (
    <ScrollView>
      <Button
        title="FILTER"
        onPress={onFilterButtonPress}
        mode="primary"
        iconName={filterOptionsExpanded ? 'up' : 'down'}
        iconColor={Colors.white}
      />
      {filterOptionsExpanded && (
        <Card style={styles.card}>
          <View>
            <Text>STATUS</Text>
            <CheckboxItem
              label="Alive"
              value="Alive"
              status={status}
              onPress={handleStatusChange}
            />
            <CheckboxItem
              label="Dead"
              value="Dead"
              status={status}
              onPress={handleStatusChange}
            />
            <CheckboxItem
              label="Unknown"
              value="Unknown"
              status={status}
              onPress={handleStatusChange}
            />
          </View>
          <View>
            <Text>SPECIES</Text>
            <CheckboxItem
              label="Human"
              value="Human"
              status={species}
              onPress={handleSpeciesChange}
            />
            <CheckboxItem
              label="Humanoid"
              value="Humanoid"
              status={species}
              onPress={handleSpeciesChange}
            />
          </View>
          <View style={styles.buttonsContainer}>
            <Button title="RESET" onPress={onResetFiltersPress} mode="white" />
            <Button title="APPLY" onPress={onApplyPress} mode="primary" />
          </View>
        </Card>
      )}
    </ScrollView>
  );
};
