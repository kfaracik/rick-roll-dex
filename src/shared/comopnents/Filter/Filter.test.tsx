import React from 'react';
import {render} from '@testing-library/react-native';
import {Filter} from './Filter';

describe('Button component', () => {
  it('should generate snapshot', () => {
    const {toJSON} = render(<Filter onApplyFilters={jest.fn} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
