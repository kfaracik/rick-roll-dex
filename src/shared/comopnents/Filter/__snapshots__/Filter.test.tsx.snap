// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should generate snapshot 1`] = `
<RCTScrollView
  style={
    {
      "marginBottom": 16,
    }
  }
>
  <View>
    <View
      collapsable={false}
      style={
        {
          "alignSelf": "flex-start",
          "backgroundColor": "#224229",
          "borderRadius": 24,
          "height": 34,
          "marginHorizontal": 16,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#224229",
            "borderColor": "#224229",
            "borderRadius": 24,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flex": 1,
            "minWidth": 64,
            "rippleColor": "#162C1B",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 24,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "margin": -6,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      {
                        "textTransform": "uppercase",
                      },
                      {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FFFFFF",
                      "fontSize": 14,
                    },
                  ],
                ]
              }
              testID="button-text"
            >
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    {
                      "color": "#FFFFFF",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                FILTER 
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "transform": [
                        {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#FFFFFF",
                          "fontSize": 10,
                        },
                        undefined,
                        {
                          "fontFamily": "anticon",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    Óòù
                  </Text>
                </View>
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Filter component interactions should generate snapshot 1`] = `
<RCTScrollView
  style={
    {
      "marginBottom": 16,
    }
  }
>
  <View>
    <View
      collapsable={false}
      style={
        {
          "alignSelf": "flex-start",
          "backgroundColor": "#224229",
          "borderRadius": 24,
          "height": 34,
          "marginHorizontal": 16,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#224229",
            "borderColor": "#224229",
            "borderRadius": 24,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flex": 1,
            "minWidth": 64,
            "rippleColor": "#162C1B",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 24,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "margin": -6,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      {
                        "textTransform": "uppercase",
                      },
                      {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FFFFFF",
                      "fontSize": 14,
                    },
                  ],
                ]
              }
              testID="button-text"
            >
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    {
                      "color": "#FFFFFF",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                FILTER 
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "transform": [
                        {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#FFFFFF",
                          "fontSize": 10,
                        },
                        undefined,
                        {
                          "fontFamily": "anticon",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    Óòù
                  </Text>
                </View>
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
